AVD updates (#373)
Azure/Azure-Proactive-Resiliency-Library: docs/content/services/specialized-workloads/azure-virtual-desktop/_index.md

## Summary of Recommendations

> {{< table style="table-striped" >}}

Reviewer: For all new recommendations, please ensure that category names align to what is specified in the contribution guide. https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing/#recommendation-categories

| [AVD-23 Ensure a unique OU when deploying VMs to Domain](#avd-23---ensure-a-unique-ou-when-deploying-vms-to-domain) | GPO | Medium | Verified | No |
| [AVD-24 Ensure the standard FSLogix configuration is deployed](#avd-24---ensure-the-standard-fslogix-configuration-is-deployed) | FSLogix Profiles | Medium | Verified | No |
| [AVD-25 Ensure user permissions are set correctly on SMB shares](#avd-25---ensure-user-permissions-are-set-correctly-on-smb-shares) | FSLogix Profiles | Medium | Verified | No |
| [AVD-26 Configure Diagnostic Settings for FSLogix logs and enable review for accounts](#avd-26---configure-diagnostic-settings-for-fslogix-logs-and-enable-review-for-accounts) | FSLogix Profiles | Medium | Verified | No |
| [AVD-27 Manually update new FSLogix image when available](#avd-27---manually-update-new-fslogix-image-when-available) | FSLogix Profiles | Low | Verified | No |
| [AVD-28 Turn on Continuous Availability for ANF if using App Attach](#avd-28---turn-on-continuous-availability-for-anf-if-using-app-attach) | App Attach Storage | Medium | Verified | No |

> | [AVD-29 App attach should be placed in separate file share; Disaster recovery plan should include App attach storage.](#avd-29---app-attach-should-be-placed-in-separate-file-share-and-disaster-recovery-plan-should-include-app-attach-storage)

Reviewer: ```suggestion
| [AVD-29 App attach should be placed in separate file share; Disaster recovery plan should include App attach storage](#avd-29---app-attach-should-be-placed-in-separate-file-share-and-disaster-recovery-plan-should-include-app-attach-storage)
```

This is a broken link

| [AVD-30 Ensure virtual networks have route tables/route server configured for all regions](#avd-30---ensure-virtual-networks-have-route-tablesroute-server-configured-for-all-regions) | Networking/Connectivity | Medium | Verified | No |
| [AVD-31 Ensure virtual networks isolation with separate IP space and NSGs for Prod and DR](#avd-31---ensure-virtual-networks-isolation-with-separate-ip-space-and-nsgs-for-prod-and-dr) | Networking/Connectivity | Medium | Verified | No |

> | [AVD-32 Ensure the network is resilient](#avd-32---ensure-the-network-is-resilient) | Networking/Connectivity | Medium | Verified | No |

Reviewer: This link is broken, please check and validate within the PR tests.

Reviewer 2: Reviewed. The URLs seem to be working.

| [VPNG-1 - Choose a Zone-redundant gateway](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/networking/vpn-gateway/#vpng-1---choose-a-zone-redundant-gateway) | Availability | High | Verified | Yes |
| [VPNG-3 - Plan for Site-to-Site VPN and Azure ExpressRoute coexisting connection](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/networking/vpn-gateway/#vpng-3---plan-for-site-to-site-vpn-and-azure-expressroute-coexisting-connection) | Disaster Recovery | High | Verified | No |
| [NSG-4 - Configure NSG Flow Logs](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/networking/network-security-group/#nsg-4---configure-nsg-flow-logs) | Monitoring | Medium | Verified | Yes |
| [VM-21 - Configure diagnostic settings for all Azure Virtual Machines](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/compute/virtual-machines/#vm-21---configure-diagnostic-settings-for-all-azure-virtual-machines) | Monitoring | Low | Verified | Yes |

> | [VM-25 - Do not create more than 2000 Citrix VDA servers per subscription](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/compute/virtual-machines/#vm-25---do-not-create-more-than-2000-citrix-vda-servers-per-subscription) | Application Resiliency | High | Verified | Yes |

Reviewer: ```suggestion
| [VM-25 - Do not create more than 2000 Citrix VDA servers per subscription](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/management/subscription/#ms-1---do-not-create-more-than-2000-citrix-vda-servers-per-subscription) | Application Resiliency | High | Verified | Yes |
```

> | [ST-1 - Ensure that Storage Account configuration is at least Zone redundant](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/storage/storage-account/#st-1---ensure-that-storage-account-configuration-is-at-least-zone-redundant) | Storage | High | Verified | Yes |

Reviewer: For all recommendations that are referenced, can you ensure fields match, specifically for the state column in this case.

Reviewer 2: Moises has applied the changes.

{{< /collapse >}}

<br><br>

### AVD-11 - Scaling plans should be created per region and not scaled across regions

**Category: Disaster Recovery**

**Impact: Medium**

**Guidance:**

> Each region has its own scaling plans assigned to host pools within that region. However, these plans can become inaccessible if there's a regional failure. To mitigate this risk, it's advisable to create a secondary scaling plan in another region

Reviewer: ```suggestion
Each region has its own scaling plans assigned to host pools within that region. However, these plans can become inaccessible if there's a regional failure. To mitigate this risk, it's advisable to create a secondary scaling plan in another region.
```

{{< code lang="sql" file="code/avd-13/avd-13.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVD-14 - Incorporate Secondary Entra ID connect synchronization server

**Category: Access & Security**

**Impact: Low**

**Guidance:**
For optimal performance and reliability, Entra ID Connect is best operated within Azure's environment, although it can be hosted on-premises if necessary. As part of a hybrid setup, it's advisable to establish secondary VMs in staging mode to prepare for failover scenarios.

To ensure continuity, set up a secondary server for Entra Connect in staging mode. This secondary server will serve as a backup, ready to sync with Entra in the event of a primary server outage.

**Resources:**

- [Learn More](https://learn.microsoft.com/en-us/entra/identity/hybrid/connect/how-to-connect-install-multiple-domains)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-14/avd-14.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVD-15 - Deploy paired Domain Controllers in the same region as AVD session hosts

**Category: Disaster Recovery**

**Impact: High**

**Guidance:**

> Ensure each Region with session host has multiple domain controllers in the same region to support adjustments based on your business requirements. To handle a large number of users, consider scaling horizontally by creating multiple host pools.

Reviewer: Please adjust guidance as `support adjustments` is not specific enough and the second sentence doesn't appear to align to this specific recommendation.

Reviewer 2: Moises has added the changes.

Hybrid - Each Azure region with AVD hosts should have ADDS DCs (IaaS VMs) in Azure and use Availability Sets or Zones for resiliency across single Datacenter or Region. This mitigates dependency on ER/VPN/Inter-Azure dependencies.

**Resources:**

- [Learn More](https://learn.microsoft.com/en-us/azure/architecture/example-scenario/azure-virtual-desktop/azure-virtual-desktop-multi-region-bcdr)

**Resource Graph Query/Scripts:**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-15/avd-15.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVD-16 - Ensure DNS regions are replicated to avoid single point of failure

**Category: Networking**

**Impact: Medium**

**Guidance:**

> ADDS Integrated DNS/other should target Secondary/Tertiary customer DNS across multi-region zones. If using custom DNS, ensure there are redundant DNS servers to avoid a single point of failure.

Reviewer: ```suggestion
Active Directory Domain Services (AD DS) integrated DNS/other should target Secondary/Tertiary customer DNS across multi-region zones. If using custom DNS, ensure there are redundant DNS servers to avoid a single point of failure.
```

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-16/avd-16.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVD-17 - Capacity Planning for AVD Resources

**Category: Disaster Recovery**

**Impact: Low**

**Guidance:**
Monitor and plan for subscription limits and API throttling limits. Closely monitor your Azure Virtual Desktop deployments and keep track of resource usage within your subscription. By proactively monitoring capacity, you can identify potential challenges early on, and you can take suitable actions to avoid reaching limits. Consider scaling across multiple subscriptions if further scaling is required, or work with Azure support to adjust limits based on your business requirements. To handle a large number of users, consider scaling horizontally by creating multiple host pools.

**Resources:**

- [Learn More](https://learn.microsoft.com/en-us/azure/architecture/example-scenario/wvd/windows-virtual-desktop#azure-virtual-desktop-limitations)

**Resource Graph Query/Scripts:**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-17/avd-17.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

> ### AVD-18 - Create new version of updated image and replace session hosts rather than update host directly

Reviewer: ```suggestion
### AVD-18 - Create updated image version and replace session hosts rather than updating host directly
```

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-35/avd-35.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVD-36 - Configure AVD Insights Workbook

**Category: Monitoring**

**Impact: High**

**Guidance**

AVD Insights is an Azure Workbook template provided by the AVD product team. It is highly recommended in order to monitor and troubleshoot AVD workloads across metrics, logs, events, and more. Both Production and DR workloads should be enabled with AVD Insights.

**Resources**

- [Learn More](https://learn.microsoft.com/en-us/azure/virtual-desktop/insights?tabs=monitor)

**Resource Graph Query/Scripts:**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-36/avd-36.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVD-37 - Ensure separate log analytics workspaces for Prod and DR

**Category: Disaster Recovery**

**Impact: Low**

**Guidance**

Having separate Log Analytics ensures that your DR environment is fully operational for visibility of the metrics, performance, and other auditing tools your workload teams will rely on in the event of an incident.

**Resources**

> - [Learn More](https://learn.microsoft.com/en-US/azure/virtual-desktop/set-up-service-alerts)

Reviewer: This link points to alerting and not log aggregation with Log Analytics.

Reviewer 2: Applied the changes.

{{< /collapse >}}

<br><br>

### AVD-11 - Scaling plans should be created per region and not scaled across regions

**Category: Disaster Recovery**

**Impact: Medium**

**Guidance:**

> Scaling plans exist per region assigned to host pools within the region and can become inaccessible in case of regional failure. A secondary scaling plan should be created with a secondary region to protect from this.

Reviewer: ```suggestion
Each region has its own scaling plans assigned to host pools within that region. However, these plans can become inaccessible if there's a regional failure. To mitigate this risk, it's advisable to create a secondary scaling plan in another region
```

Reviewer: A little bit confusing on the sentence structure, could you take a look at my suggestion and/or reword for clarity?

**Resources:**

- [Learn More](https://learn.microsoft.com/en-us/azure/virtual-desktop/autoscale-scaling-plan?tabs=portal)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-11/avd-11.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

> ### AVD-12 - Keep detailed documentation of all AVD and shared services based on BCDR strategy detailing failover scenarios covered based on agreements

Reviewer: Can we reference this recommendation in place of this: https://azure.github.io/Azure-Proactive-Resiliency-Library/well-architected/2-design/#wads-7---design-a-bcdr-strategy-that-will-help-to-meet-the-business-requirements

Reviewer 2: We have included WADS-7 recommendation in addition to the above, would it be better if we retain the above since it's AVD specific?

**Category: Disaster Recovery**

**Impact: Low**

**Guidance:**
It's essential for all stakeholders to have access to the latest documented business continuity and disaster recovery considerations aligned to the AVD workloads you deploy.

**Resources:**

- [Learn More](https://learn.microsoft.com/en-us/azure/virtual-desktop/network-connectivity)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-12/avd-12.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVD-13 - Validate that the AVD session hosts can communicate with the AVD control plane and UDP ports are open if UDP is in use

**Category: User Connection**

**Impact: Medium**

**Guidance:**
Connectivity for validating the VMs connectivity to AVD Control Plane and UDP TURN ports open.

> Global URLs confirmed whitelisted. UDP/TURN ports open and accessible.

Reviewer: Can you please reword for clarity as it's a little unclear for me what with how the sentences are worded.

**Resources:**

- [Learn More](https://learn.microsoft.com/en-us/azure/virtual-desktop/troubleshoot-rdp-shortpath)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-13/avd-13.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

> ### AVD-14 - Ensure Secondary Entra ID connect synchronization server

Reviewer: ```suggestion
### AVD-14 - Incorporate Secondary Entra ID connect synchronization server
```

**Category: Identity/Directory**

**Impact: Low**

**Guidance:**

> Hybrid - Entra ID Connect best to run in Azure but can be hosted on-prem. Secondary or more VMs should be setup in staging mode in event of failover.
> Set up secondary server in staging mode for Entra Connect for syncing to Entra in case of primary server outage.

Reviewer: ```suggestion
For optimal performance and reliability, Entra ID Connect is best operated within Azure's environment, although it can be hosted on-premises if necessary. As part of a hybrid setup, it's advisable to establish secondary VMs in staging mode to prepare for failover scenarios.

To ensure continuity, set up a secondary server for Entra Connect in staging mode. This secondary server will serve as a backup, ready to sync with Entra in the event of a primary server outage.
```

**Resources:**

- [Learn More](https://learn.microsoft.com/en-us/entra/identity/hybrid/connect/how-to-connect-install-multiple-domains)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-14/avd-14.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVD-15 - Deploy paired Domain Controllers in the same region as AVD session hosts

**Category: Disaster Recovery**

**Impact: High**

**Guidance:**
Ensure each Region with session host has multiple domain controllers in the same region to support adjustments based on your business requirements. To handle a large number of users, consider scaling horizontally by creating multiple host pools.

> Hybrid - Each Azure region with AVD hosts should have ADDS DCs (IaaS VMs) in Azure and use Availability Sets or Zones for resiliency across single Datacenter or Region. Mitigates dependency on ER/VPN/Inter-Azure dependencies.

Reviewer: ```suggestion
Hybrid - Each Azure region with AVD hosts should have ADDS DCs (IaaS VMs) in Azure and use Availability Sets or Zones for resiliency across single Datacenter or Region. This mitigates dependency on ER/VPN/Inter-Azure dependencies.
```

To ensure your apps work with the latest updates, the validation host pool should be as similar to host pools in your non-validation environment as possible. Users should connect as frequently to the validation host pool as they do to the standard host pool. If you have automated testing on your host pool, you should include automated testing on the validation host pool.

**Resources:**

- [Learn More](https://learn.microsoft.com/en-us/azure/virtual-desktop/configure-validation-environment?tabs=azure-portal)

**Resource Graph Query/Scripts:**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avd-19/avd-19.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVD-20 - [Pooled] Configure scheduled agent updates

**Category: Pooled Session Host VMs**

**Impact: Medium**

**Guidance:**
Check if schedules have been created to provide maintenance windows.

> Scheduled AVD agent updates on the host pool.

Reviewer: Is this sentence needed or could it be incorporated differently?
